src/datasources/asset/components/AssetQueryEditor.tsx

```diff
@@ -13,73 +13,83 @@ import { defaultAssetSummaryQuery, defaultCalibrationForecastQuery, defaultListA
+      calibrationForecastEnabled: datasource.instanceSettings.jsonData?.calibrationForecastEnabled ?? false,
+      assetSummaryEnabled: datasource.instanceSettings.jsonData?.assetSummaryEnabled ?? false,
+    });
+  }, [datasource.instanceSettings.jsonData]);
 
   const handleQueryChange = useCallback((value: AssetQuery, runQuery = false): void => {
-    onChangeRef.current({ ...queryRef.current, ...value });
+    onChange({ ...query, ...value });
     if (runQuery) {
-      onRunQueryRef.current();
+      onRunQuery();
     }
-  }, []);
+  }, [query, onChange, onRunQuery]);
 
-  const handleQueryTypeChange = (item: SelectableValue<AssetQueryType>): void => {
+  const handleQueryTypeChange = useCallback((item: SelectableValue<AssetQueryType>): void => {
     setQueryType(item.value!);
-  };
+    handleQueryChange({ ...query, queryType: item.value! }, true);
+  }, [query, handleQueryChange]);
 
   useEffect(() => {
-    if (queryType === AssetQueryType.ListAssets) {
-      handleQueryChange({ ...queryRef.current, queryType: AssetQueryType.ListAssets, ...defaultListAssetsQuery }, true);
+    if(queryType === query.queryType) {
+      return;
     }
-    if (queryType === AssetQueryType.CalibrationForecast) {
-      handleQueryChange({
-        ...queryRef.current,
-        queryType: AssetQueryType.CalibrationForecast,
-        ...defaultCalibrationForecastQuery,
-      });
+
+    if (queryType === AssetQueryType.ListAssets && assetFeatures.assetListEnabled) {
+      handleQueryChange({ ...query, queryType: AssetQueryType.ListAssets, ...defaultListAssetsQuery }, true);
+    }
+    if (queryType === AssetQueryType.CalibrationForecast && assetFeatures.calibrationForecastEnabled) {
+      handleQueryChange({ ...query, queryType: AssetQueryType.CalibrationForecast, ...defaultCalibrationForecastQuery }, true);
     }
-    if (queryType === AssetQueryType.AssetSummary) {
-      handleQueryChange({ ...queryRef.current, queryType: AssetQueryType.AssetSummary, ...defaultAssetSummaryQuery });
+    if (queryType === AssetQueryType.AssetSummary && assetFeatures.assetSummaryEnabled) {
+      handleQueryChange({ ...query, queryType: AssetQueryType.AssetSummary, ...defaultAssetSummaryQuery }, true);
     }
-  }, [queryType, handleQueryChange]);
+  }, [
+    query, queryType, handleQueryChange, onRunQuery,
+    assetFeatures.assetListEnabled, assetFeatures.calibrationForecastEnabled, assetFeatures.assetSummaryEnabled
+  ]);
+
+  const filterOptions = useMemo(() => {
+    return queryTypeOptions.filter(option => {
+      return (option.value === AssetQueryType.ListAssets && assetFeatures.assetListEnabled) ||
+        (option.value === AssetQueryType.CalibrationForecast && assetFeatures.calibrationForecastEnabled) ||
+        (option.value === AssetQueryType.AssetSummary && assetFeatures.assetSummaryEnabled)
+    });
+  }, [assetFeatures.assetListEnabled, assetFeatures.calibrationForecastEnabled, assetFeatures.assetSummaryEnabled]);
```

Comment:
Have we thought about what happens if a user creates a query hidden behind a toggle and then loads the dashboard without the toggle enabled? We may want to check the type of the incoming query and display the option even if the toggle is disabled.

Reply:
Yes I've thought of it: 

1) The widget query should remain unchanged.
2) When trying to edit the values of a DataSource functionality that has been disabled, we should only show the selection without the possibility of editing the configured values.